src/libsync/clientsideencryption.cpp

```diff
@@ -828,6 +867,31 @@ bool ClientSideEncryption::checkPublicKeyValidity(const AccountPtr &account) con
     return true;
 }
 
+bool ClientSideEncryption::checkServerPublicKeyValidity(const QString &serverPublicKeyString) const
+{
+    Bio serverPublicKeyBio;
+    auto serverPublicKeyPem = serverPublicKeyString.toLatin1();
+    BIO_write(serverPublicKeyBio, serverPublicKeyPem.constData(), serverPublicKeyPem.size());
+    auto serverPublicKey = PKey::readPrivateKey(serverPublicKeyBio);
+
+    Bio certificateBio;
+    auto certificatePem = _certificate.toPem();
+    BIO_write(certificateBio, certificatePem.constData(), certificatePem.size());
+    auto x509Certificate = X509Certificate::readCertificate(certificateBio);
+    if (!x509Certificate) {
+        qCInfo(lcCse()) << "Client certificate is invalid. Impossible to check it against the server public key";
```

Comment:
@mgallien `Impossible` is a nice word, though, `Could not` would sound more correct :)